server/superdesk/eve_backend.py

```diff
@@ -120,13 +123,37 @@ def replace_in_search(self, endpoint_name, id, document, original):
             search_backend.replace(endpoint_name, id, document)
 
     def delete(self, endpoint_name, lookup):
+        """
+        Delete method to delete by using mongo query syntax
+        :param endpoint_name: Name of the endpoint
+        :param lookup: User mongo query syntax. example 1. {'_id':123}, 2. {'item_id': {'$in': [123, 234]}}
+        :returns:
+        Returns the mongo remove command response. {'n': 12, 'ok': 1}
+        """
         backend = self._backend(endpoint_name)
-        res = backend.remove(endpoint_name, lookup)
         search_backend = self._lookup_backend(endpoint_name)
-        if search_backend is not None:
-            search_backend.remove(endpoint_name, lookup)
+        docs = self.get_from_mongo(endpoint_name, lookup=lookup, req=ParsedRequest())
+        ids = [doc[config.ID_FIELD] for doc in docs]
+        res = backend.remove(endpoint_name, {config.ID_FIELD: {'$in': ids}})
+        if res and res.get('n', 0) > 0 and search_backend is not None:
+            self._remove_documents_from_search_backend(endpoint_name, ids)
         return res
 
+    def _remove_documents_from_search_backend(self, endpoint_name, ids):
+        """
+        remove documents from search backend.
+        :param endpoint_name: name of the endpoint
+        :param ids: list of ids
+        """
+        ids = [str(doc_id) for doc_id in ids]
+        batch_size = 500
+        logger.info("total documents to be removed {}".format(len(ids)))
+        for i in range(0, len(ids), batch_size):
+            batch = ids[i:i + batch_size]
+            query = {'query': {'terms': {'{}._id'.format(endpoint_name): batch}}}
+            app.data._search_backend(endpoint_name).remove(endpoint_name, query)
```

Comment:
Is eve-elastic returning any result after deletion? We should be able to compare the number of deleted items and log a warning/error if they don't match...